service/src/CpufreqSysfsDriver.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include "CpufreqSysfsDriver.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "SysfsIOGroup.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm_topo.h"
+
+#include <cmath>
+#include <cstring>
+#include <sstream>
+
+// Arbitrary buffer size. We're generally looking at integer values much shorter
+// than 100 digits in length. The IOGroup performs string truncation checks in
+// case that ever changes.
+static const std::string CPUFREQ_DIRECTORY = "/sys/devices/system/cpu/cpufreq";
+
+static std::map<int, std::string> load_cpufreq_resources_by_cpu(const std::string &cpufreq_directory)
+{
+    char cpu_buf[geopm::SysfsDriver::M_IO_BUFFER_SIZE] = {0};
+    std::map<int, std::string> result;
+
+    for (const auto &policy_file : geopm::list_directory_files(cpufreq_directory)) {
+        if (policy_file.find("policy") != 0) {
+            continue;
+        }
+
+        std::ostringstream oss;
+        oss << cpufreq_directory << "/" << policy_file;
+        auto resource_path = oss.str();
+        oss << "/affected_cpus";
+        auto cpu_map_path = oss.str();
+
+        int cpu_map_fd = open(cpu_map_path.c_str(), O_RDONLY);
+        if (cpu_map_fd == -1) {
+            throw geopm::Exception("SysfsIOGroup failed to open " + cpu_map_path,
```

Comment:
```suggestion
            throw geopm::Exception("CpufreqSysfsDriver failed to open " + cpu_map_path,
```
I see that this function is called in the constructor of CpufreqSysfsDriver.  Other exception messages from the class start this way.  Unclear if this was intentional or not.  If not, I think we should change all the exceptions in this func to start with "CpufreqSysfsDriver".